propkatraj/propkatraj.py

```diff
@@ -210,161 +208,22 @@ def _single_frame(self):
-                " for more details")
-        warnings.warn(wmsg)
-
-    # "filename" for our stream
-    # use same name so that propka overwrites
-    try:
-        newname = os.path.join(os.path.dirname(universe.filename),
-                               'current.pdb')
-    except TypeError:
-        # we have a trajectory without a directory
-        newname = os.path.join(os.path.curdir, 'current.pdb')
-
-    # progress logging output (because this is slow...)
-    pm = mda.lib.log.ProgressMeter(universe.trajectory.n_frames,
-                                   format="{step:5d}/{numsteps} t={time:12.3f} ps  "
-                                   "[{percentage:5.1f}%]",
-                                   interval=1)
-
-    times = []
-    pkas = []
-    failed_frames = 0
-    failed_frames_log = []
-    for ts in universe.trajectory[start:stop:step]:
-        pm.echo(ts.frame, time=ts.time)
-
-        # we create a named stream to write the atoms of interest into
-        pstream = mda.lib.util.NamedStream(StringIO(), newname)
-        atomsel.write(pstream)
-
-        pstream.reset()         # reset for reading
-
-        # we feed the stream to propka, and it reads it as if it were a file on
-        # disk
-        try:
-            mol = pk.single(pstream, optargs=['--quiet'])
-        except (IndexError, AttributeError) as err:
-            # https://github.com/Becksteinlab/propkatraj/issues/13
-            # https://github.com/Becksteinlab/propkatraj/issues/10
-            err_msg = "{0} (failure {2}): failing frame {1}".format(
-                    universe.trajectory.filename, ts.frame, failed_frames)
-            if not skip_failure:
-                raise_from(RuntimeError(err_msg), err)
-            else:
-                failed_frames += 1
-                failed_frames_log.append(ts.frame)
-                logging.warning(err_msg)
-                continue
-        finally:
-            pstream.close(force=True)  # deallocate
-
-        # parse propka data structures to get out what we actually want
-        confname = mol.conformation_names[0]
-        conformation = mol.conformations[confname]
-        groups = conformation.get_titratable_groups()
-
-        # extract pka estimates from each residue
-        pkas.append([g.pka_value for g in groups])
-
-        # record time
-        times.append(ts.time)
-
-    if failed_frames_log:
-        logging.warning('number of failed frames = {0}'.format(failed_frames))
-        logging.warning('percent failure = {0:.3f}%'.format(
-            float(failed_frames)/len(universe.trajectory)*100))
-        logging.warning('failed frames: %r', failed_frames_log)
-
-    # a `pandas.DataFrame` is a good data structure for this data
-    df = pd.DataFrame(pkas, index=pd.Float64Index(times, name='time'),
-                      columns=[g.atom.resNumb for g in groups])
-
-    return df
+        self.results.pkas = pd.DataFrame(self._pkas, index=pd.Index(self.times,
```

Comment:
update docs that pkas are in `results.pkas`

Reply:
done